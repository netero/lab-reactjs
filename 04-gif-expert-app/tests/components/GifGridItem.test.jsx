const { render, screen } = require("@testing-library/react");
const { GifGridItem } = require("../../src/components/GifGridItem");

describe('Pruebas en <GifItem />',()=>{
    const title ='Title';
    const url ='http://url.cl/';
    
    test('Debe hacer match con el snapshot',()=>{
        const { container } = render( <GifGridItem url={url} title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifGridItem url={url} title={title}/>);
        //screen.debug();
        const { src, alt } = screen.getByRole('img');
        expect (src).toBe(url);
        expect (alt).toBe(title);
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render( <GifGridItem url={url} title={title}/>);
        expect( screen.getByText( title )).toBeTruthy();
    });
});