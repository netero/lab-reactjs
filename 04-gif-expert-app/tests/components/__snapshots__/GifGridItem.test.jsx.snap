// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="http://url.cl/"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
